packages/uicore/src/components/GroupedThumbnailSelect/GroupedThumbnailSelect.tsx

```diff
@@ -45,19 +45,20 @@ export function GroupedThumbnailSelect(props: ConnectedGroupedThumbnailSelectPro
     cancelText = 'Close',
     className,
     thumbnailClassName,
+    expandAllByDefault,
     onChange
   } = props
   const value = get(formik.values, name)
 
-  const [showAllOptions, setShowAllOptions] = React.useState(isEmpty(value))
+  const [showAllOptions, setShowAllOptions] = React.useState(isEmpty(value) || expandAllByDefault)
   const [visibleGroups, setVisibleGroups] = React.useState(groups)
 
   const hasError = errorCheck(name, formik)
   const intent = hasError ? Intent.DANGER : Intent.NONE
   const helperText = hasError ? <FormError name={name} errorMessage={get(formik?.errors, name)} /> : null
 
   React.useEffect(() => {
-    setShowAllOptions(isEmpty(value))
+    setShowAllOptions(isEmpty(value) || expandAllByDefault)
   }, [value])
```

Comment:
We can add `expandAllByDefault` as a dependency

Reply:
adding in next PR